### PR TITLE
Sul23 434

### PR DESCRIPTION
**Pull request by imonroe**
# READY FOR REVIEW

# Summary
- The `stanford-lists` component needed to be made aware of the shared tags event view we set up here: https://github.com/SU-SWS/ace-stanfordlagunita/pull/98

# Review By (Date)
- 6/5/2024, prior to release

# Urgency
- the new view won't work without it.

# Steps to Test

1.  I tested using the branner-earth-sciences-library-map-collections location, since they have a card set up for events using the old shared-tags view.  
2. Create a new event if necessary, scheduled at some point in the future.  
3. When you set up the event, added the Branner Library location in the taxonomy fields.
4. in the Branner LIbrary location page, find the Events list card.  In that card, change the view from `"Shared Tags" Content` to `Shared Tag Events`.  No other change should be necessary.
5. On the front end, make sure that only events which are scheduled today or later are pulled in to the list card, and make sure it still looks correct. 

# Affected Projects or Products
- no

# Associated Issues and/or People
- https://github.com/SU-SWS/ace-stanfordlagunita/pull/98

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

File: src/components/paragraph/stanford-lists.tsx [hunk 1]
```diff
@@ -140,6 +144,12 @@ const getViewPagedItems = cache(async (viewId: string, displayId: string, contex
       items = graphqlResponse.stanfordSharedTags?.results as unknown as NodeUnion[]
       break
 
+    case 'sul_shared_tag_events--card_grid':
```
Comment (imonroe) [inline src/components/paragraph/stanford-lists.tsx:147]: I'm not sure if this is necessary or not.

File: src/components/paragraph/stanford-lists.tsx [hunk 2]
```diff
@@ -101,6 +101,10 @@ const getViewPagedItems = cache(async (viewId: string, displayId: string, contex
       tags.push('views:all');
       break;
 
+    case 'sul_shared_tag_events--card_grid':
+      tags.push('views:all')
```
Comment (pookmish) [inline src/components/paragraph/stanford-lists.tsx:105]: use `stanford_events`, not `all`

File: src/components/paragraph/stanford-lists.tsx [hunk 3]
```diff
@@ -140,6 +144,12 @@ const getViewPagedItems = cache(async (viewId: string, displayId: string, contex
       items = graphqlResponse.stanfordSharedTags?.results as unknown as NodeUnion[]
       break
 
+    case 'sul_shared_tag_events--card_grid':
+      filters = getViewFilters(['term_node_taxonomy_name_depth', 'type'], contextualFilter)
+      graphqlResponse = await client.stanfordSharedTags({filters, ...queryVariables});
```
Comment (pookmish) [inline src/components/paragraph/stanford-lists.tsx:149]: wrong query. See the `https://github.com/SU-SWS/sulgryphon-nextjs/blob/1.x/src/lib/gql/view-queries.drupal.gql` to start

File: src/components/paragraph/stanford-lists.tsx (hunk 1 above)
Comment (pookmish) [reply src/components/paragraph/stanford-lists.tsx:147]: yes it definitely is needed.

**Review by pookmish — CHANGES_REQUESTED** [review]

File: src/lib/gql/__generated__/drupal.d.ts [hunk 4]
```diff
@@ -2512,6 +2782,70 @@ export type QueryTermArgs = {
   langcode?: InputMaybe<Scalars['String']['input']>;
 };
 
+/** Entity type redirect. */
```
Comment (imonroe) [inline src/lib/gql/__generated__/drupal.d.ts:2785]: Could this have something to do with why redirects don't seem to take effect correctly when made on the Drupal side?

File: src/components/paragraph/stanford-lists.tsx (hunk 2 above)
Comment (imonroe) [reply src/components/paragraph/stanford-lists.tsx]: fixed.

**Review by pookmish — APPROVED** [review]
G2g now

**Pull request merged**